### PR TITLE
Adjust vcd_nsxt_alb_settings to support IPv6 and Transparent mode

### PR DESCRIPTION
**Pull request by Didainius**
This PRs builds on top of https://github.com/vmware/go-vcloud-director/pull/549 and adds two new fields to `vcd_nsxt_alb_settings` resource and data source:
* `ipv6_service_network_specification` (requires VCD 10.4.0+). 
* `is_transparent_mode_enabled` (requires VCD 10.4.1+). This field allows using a "Preserve Client IP" on a Virtual Service. Note. The Virtual Service resource (`vcd_nsxt_alb_virtual_service`) will be updated in separate PRs during this release cycle.

### CODE REVIEW

File: .changes/v3.9.0/XXX-improvements.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+* `vcd_nsxt_alb_settings` resource and data source adds two new fields `is_transparent_mode_enabled`
+  and `ipv6_service_network_specification` [GH-XXX]
```
Comment (adambarreiro) [inline .changes/v3.9.0/XXX-improvements.md:2]:
```suggestion
  and `ipv6_service_network_specification` [GH-996]
```
(Same with file name)

File: website/docs/r/nsxt_alb_settings.html.markdown [hunk 2]
```diff
@@ -46,10 +68,20 @@ The following arguments are supported:
...
   will set it to `false`
+* `supported_feature_set` - (Optional; *v3.7+*) Feature set of this Edge Gateway if ALB is enabled (`STANDARD` or `PREMIUM`)
+* `is_transparent_mode_enabled` - (Optional; *v3.9+*, *VCD 10.4.1+*) When enabled, it allows to
+  configure Preserve Client IP on a Virtual Service
 * `service_network_specification` - (Optional) Gateway CIDR format which will be used by Load Balancer service. All the
   load balancer service engines associated with the Service Engine Group will be attached to this network. The subnet
   prefix length must be 25. If nothing is set, the **default is 192.168.255.125/25**. This field cannot be updated
-* `supported_feature_set` - (Optional; *v3.7+*) Feature set of this Edge Gateway if ALB is enabled (`STANDARD` or `PREMIUM`)
+* `ipv6_service_network_specification` (Optional; *v3.9+*, *VCD 10.4.0+*) The IPv6 network definition in Gateway
+  CIDR format which will be used by Load Balancer service on Edge. All the load balancer service
```
Comment (adambarreiro) [inline website/docs/r/nsxt_alb_settings.html.markdown:78]: Maybe
```suggestion
* `ipv6_service_network_specification` (Optional; *v3.9+*, *VCD 10.4.0+*) The IPv6 network definition in
  CIDR format which will be used by Load Balancer service in the Edge Gateway. All the load balancer service
```

**Review by adambarreiro — COMMENTED** [review]
First scan

File: website/docs/r/nsxt_alb_settings.html.markdown (hunk 2 above)
Comment (Didainius) [reply website/docs/r/nsxt_alb_settings.html.markdown:78]: That works. Changed.

File: .changes/v3.9.0/XXX-improvements.md (hunk 1 above)
Comment (Didainius) [reply .changes/v3.9.0/XXX-improvements.md]: You got me here. Changed :)

File: website/docs/r/nsxt_alb_settings.html.markdown [hunk 3]
```diff
@@ -46,10 +68,20 @@ The following arguments are supported:
...
   prefix length must be 25. If nothing is set, the **default is 192.168.255.125/25**. This field cannot be updated
-* `supported_feature_set` - (Optional; *v3.7+*) Feature set of this Edge Gateway if ALB is enabled (`STANDARD` or `PREMIUM`)
+* `ipv6_service_network_specification` (Optional; *v3.9+*, *VCD 10.4.0+*) The IPv6 network
+  definition in CIDR format which will be used by Load Balancer service in the Edge Gateway. All the
+  load balancer service engines associated with the Service Engine Group will be attached to this
+  network. This field cannot be updated
+
+~> IPv4 service network will be used if both the `service_network_specification` and
+`ipv6_service_network_specification` properties are unset. If both are set, it will still be one
+service network with a dual IPv4 and IPv6 stack.
```
Comment (lvirbalas) [inline website/docs/r/nsxt_alb_settings.html.markdown:84]: And what if `service_network_specification` is unset, but `ipv6_service_network_specification` is set?

Comment (Didainius) [reply website/docs/r/nsxt_alb_settings.html.markdown:84]: Good question, which is not directly documented anywhere, but it uses only IPv6 then as per UI. I have put an explicit test step for that as well.

Setting only IPv6 in UI
image

Results in such configuration as well

image

**Review by lvirbalas — APPROVED** [review]

**Review by dataclouder — APPROVED** [review]

**Review by adezxc — APPROVED** [review]

File: vcd/resource_vcd_nsxt_alb_settings_test.go [hunk 4]
```diff
@@ -200,6 +200,256 @@ resource "vcd_nsxt_alb_settings" "test" {
...
+		Steps: []resource.TestStep{
+			{
+				Config: configText1, // Setup prerequisites - configure NSX-T ALB in Provider
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestMatchResourceAttr("vcd_nsxt_alb_controller.first", "id", regexp.MustCompile(`\d*`)),
+					resource.TestMatchResourceAttr("vcd_nsxt_alb_cloud.first", "id", regexp.MustCompile(`\d*`)),
+					resource.TestMatchResourceAttr("vcd_nsxt_alb_service_engine_group.first", "id", regexp.MustCompile(`\d*`)),
+					resource.TestMatchResourceAttr("data.vcd_nsxt_alb_importable_cloud.cld", "id", regexp.MustCompile(`\d*`)),
+					resource.TestCheckResourceAttr("vcd_nsxt_alb_settings.test", "service_network_specification", "10.10.255.225/27"),
+					resource.TestCheckResourceAttr("vcd_nsxt_alb_settings.test", "ipv6_service_network_specification", "2001:0db8:85a3:0000:0000:8a2e:0370:7334/120"),
```
[the remark this answers is not in the document]
Comment (Didainius) [reply vcd/resource_vcd_nsxt_alb_settings_test.go:376]: Nope - it works and returns this as expected

**Pull request merged**